scripts/install_KLU_Sundials.py

```diff
@@ -1,3 +1,15 @@
+# /// pyproject
+# [run]
+# requires-python = "<=3.9"
```

Comment:
This should be Python 3.8–3.12 based on the Python version compatibility we currently provide

Reply:
> This should be Python 3.8–3.12 based on the Python version compatibility we currently provide

Thank you for your review. I've updated the Python version compatibility range in the `install_KLU_Sundials.py` script.

The commit with these updates can be found [here](https://github.com/pybamm-team/PyBaMM/pull/3716/commits/0bd42a59e4d65ffdd393cd576ce758ce094d7499)